Source/Microsoft.Teams.Apps.FAQPlusPlus/Cards/ResponseCard.cs

```diff
@@ -18,75 +19,217 @@ namespace Microsoft.Teams.Apps.FAQPlusPlus.Cards
+        /// <param name="answer">The answer string.</param>
+        /// <returns>A list of adaptive actions.</returns>
+        private static List<AdaptiveAction> BuildListOfActions(string userQuestion, string answer)
+        {
+            List<AdaptiveAction> actionsList = new List<AdaptiveAction>
+            {
+                // Adds the "Ask an expert" button.
+                new AdaptiveSubmitAction
+                {
+                    Title = Strings.AskAnExpertButtonText,
+                    Data = new ResponseCardPayload
+                    {
+                        MsTeams = new CardAction
+                        {
+                            Type = ActionTypes.MessageBack,
+                            DisplayText = Strings.AskAnExpertDisplayText,
+                            Text = Constants.AskAnExpert,
+                        },
+                        UserQuestion = userQuestion,
+                        KnowledgeBaseAnswer = answer,
+                    },
+                },
+
+                // Adds the "Share feedback" button.
+                new AdaptiveSubmitAction
+                {
+                    Title = Strings.ShareFeedbackButtonText,
+                    Data = new ResponseCardPayload
+                    {
+                        MsTeams = new CardAction
+                        {
+                            Type = ActionTypes.MessageBack,
+                            DisplayText = Strings.ShareFeedbackDisplayText,
+                            Text = Constants.ShareFeedback,
+                        },
+                        UserQuestion = userQuestion,
+                        KnowledgeBaseAnswer = answer,
                     },
                 },
             };
 
-            return new Attachment
+            return actionsList;
+        }
+
+        /// <summary>
+        /// This method will build the list of previous questions.
+        /// </summary>
+        /// <param name="id">The QnA Id of the previous question.</param>
+        /// <param name="userQuestion">The question that was asked by the user originally.</param>
+        /// <param name="answer">The knowledge base answer.</param>
+        /// <param name="payload">The response card payload.</param>
+        /// <returns>A list of previous questions.</returns>
+        private static List<QnADTO> BuildListOfPreviousQuestions(int id, string userQuestion, string answer, ResponseCardPayload payload)
+        {
+            if (payload.PreviousQuestions?.Count > 0)
             {
-                ContentType = AdaptiveCard.ContentType,
-                Content = responseCard,
-            };
+                payload.PreviousQuestions.Add(new QnADTO
+                {
+                    Id = id,
+                    Questions = new List<string>()
+                    {
+                        userQuestion,
+                    },
+                    Answer = answer,
```

Comment:
Where is this answer used?

Reply:
Added answer in case to handle Ask an expert functionality which might be implemented in next release. In current version, it is just added as placeholder.